### PR TITLE
Added Unit Tests for serialze a Dictionary, Array and JsonElement

### PR DESCRIPTION
**Pull request by Brianx03**
Added Unit Tests for serialze a Dictionary, Array and JsonElement

### CODE REVIEW

File: test/Swan.Test/JsonFormatterDynamicTest.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public void WithJsonText_ReturnsObject()
+    {
+        var jsonText = "{\r\n    \"Name\": \"Merlina\"\r\n}";
+        var jsonObject = jsonText.JsonDeserialize();
+
+        Assert.AreEqual("Merlina", jsonObject?.Name);
+    }
+
+    [Test]
+    public async Task WithNullStream_ReturnsObject()
```
Comment (AntonioGGovea) [inline test/Swan.Test/JsonFormatterDynamicTest.cs:30]: Can you change the name of the test pls, it says that you are testing that it returns an object but instead you are testing that it returns null

**Review by AntonioGGovea — CHANGES_REQUESTED** [review]

File: test/Swan.Test/CsvDictionaryReaderTest.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        dictionaryReader.AddMappings(dict);
+
+        dictionaryReader.Current.TryGetValue("Company", out var addedMap);
+        
+        Assert.AreEqual(company, addedMap);
+    }
+
+    [Test]
+    public void WithHeadingMame_RemovesMapings()
```
Comment (ClementeZazueta) [inline test/Swan.Test/CsvDictionaryReaderTest.cs:43]: There are typos `Mame` and `Maping` on the name of the function, I think it should be `WithHeadingName_RemovesMappings`

File: test/Swan.Test/JsonFormatterDynamicTest.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        Stream jsonStream = null;
+        var jsonObject = await jsonStream.JsonDeserializeAsync();
+
+        Assert.IsNull(jsonObject);
+    }
+
+    [Test]
+    public async Task WithUTF8JsonStream_ReturnsDinamicObject()
```
Comment (ClementeZazueta) [inline test/Swan.Test/JsonFormatterDynamicTest.cs:39]: Typo on word `Dinamic` should be `Dynamic`

File: test/Swan.Test/JsonFormatterDynamicTest.cs (hunk 1 above)
Comment (ClementeZazueta) [reply test/Swan.Test/JsonFormatterDynamicTest.cs:30]: Also you are validating a string with `Name` prop, if you want to check an object, you should use `typeof(jsonObject)`, maybe it is correct as it is, just rename the function, is not doing what you are saying, keep in mind the `Principle of Least Surprise` 😁

File: test/Swan.Test/JsonFormatterDynamicTest.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        var dog = new Dog() { Name = "Merlina" };
+        byte[] jsonUtf8Bytes = JsonSerializer.SerializeToUtf8Bytes(dog);
+
+        var json = JsonSerializer.Serialize(dog);
+
+        var stream = new MemoryStream(jsonUtf8Bytes);
+
+        var deserializedDog = await stream.JsonDeserializeAsync();
+
+        Assert.AreEqual(dog.Name, deserializedDog?.Name);
```
Comment (ClementeZazueta) [inline test/Swan.Test/JsonFormatterDynamicTest.cs:50]: Same here, you are validating a value from a property, not an object 👀, there are more cases like this one along the code.

File: test/Swan.Test/PaginatorTest.cs [hunk 5]
```diff
@@ -46,4 +46,11 @@ public void WithValidDataAtIndexNine_ReturnGetLastItemIndex()
         var stu = new Paginator(99, 10);
         Assert.AreEqual(98, stu.GetLastItemIndex(9));
     }
+
+    [Test]
+    public void WithTotalCounZero_ReturnPageCountEqualsZero()
```
Comment (ClementeZazueta) [inline test/Swan.Test/PaginatorTest.cs:51]: Typo `Coun` should be `Count`

**Review by ClementeZazueta — CHANGES_REQUESTED** [review]
I left some comments for you man. 😄

**Review by ClementeZazueta — APPROVED** [review]

**Review by AntonioGGovea — APPROVED** [review]

**Pull request merged**